### PR TITLE
feat: add ADR doc for recommendations plugin

### PR DESCRIPTION
**Pull request by ahtesham-quraish**
Description:
Added ADR document for learner recommendations plugin

Its related to: [VAN-1396](https://2u-internal.atlassian.net/browse/VAN-1396)

### CODE REVIEW

File: docs/decisions/0001-purpose-of-this-repo.rst [hunk 1]
```diff
@@ -19,39 +20,29 @@ Status
 Context
 *******
 
-TODO: Add context of what led to the creation of this repo.
+The recommendations are shown multiple areas in edx like dashboard page, course about page and will be shown other areas as well in future. 
+The recommendations are only used by edx not by openedx community so someone from openedx may not need it.
+Currently multiple recommendations experiments are on the way which makes the LMS code tightly coupled with it. So keeping all of this in mind this repo will be
```
Comment (shahbaz-arbisoft) [inline docs/decisions/0001-purpose-of-this-repo.rst:25]: Currently, multiple recommendations experiments are on the way which makes the LMS

File: docs/decisions/0001-purpose-of-this-repo.rst [hunk 2]
```diff
@@ -1,57 +1,50 @@
-0001 Purpose of This Repo
+Django plugin for learner recommendations
 #########################
```
Comment (robrap) [inline docs/decisions/0001-purpose-of-this-repo.rst:2]: Nit: updating underline length to match.

**Review by robrap — APPROVED** [review]
Thank you. This looks fine as-is. 

That said, a couple of things to consider:
1. Docs are imperfect regarding how to best interact with edx-platform from inside a plugin, and those docs are slowly being improved. For example, you'll generally want to avoid edx-platform imports from within the plugin, and want to avoid imports of plugin code from within edx-platform. That said, this isn't always possible, and each case requires discussion about how to use a proper API, or create a proper API, or where a temporary hack is required with a future option of improving the API. You don't necessarily need to change anything, but you could add something about following and developing best practices for integration between the plugin and edx-platform. Maybe all those touchpoints are already well known?
2. Related, I don't know if you've already discussed the topic of recommendations with the community? You don't have to change anything about your current plan necessarily, but a discussion may open the door to different design decisions in the future to enable interfaces for recommendations that might have different implementations, this being one of them. A ticket could be created https://github.com/orgs/openedx/projects/4/views/1 if you wished to open up these conversations. Note that opening this up would not need to block any of your work, but could open the doors if you wish to have additional input.

**Comment by zainab-amir** [comment]
@robrap 

> For example, you'll generally want to avoid edx-platform imports from within the plugin

Regarding the above statement, what is the best practice?

> A ticket could be created https://github.com/orgs/openedx/projects/4/views/1 if you wished to open up these conversations. 

I'll get this added to the roadmap for H2 2023. I am hoping we will reach a stable state by then where we have fully transitioned to this plugin for all recommendation-related matters. This will enable us to engage in more productive discussions and explore the possibility of aligning our efforts with community interests, provided there is a genuine interest.

**Comment by robrap** [comment]
@zainab-amir: Here is a doc that is currently being updated, but provides some answers around the best practices: https://github.com/openedx/edx-platform/blob/kdmccormick/extension-points-doc/docs/guides/extension_points.rst. Feel free to ask if it doesn't provide enough clarity.

File: docs/decisions/0001-purpose-of-this-repo.rst (hunk 2 above)
Comment (ahtesham-quraish) [reply docs/decisions/0001-purpose-of-this-repo.rst]: addressed

**Pull request merged**